### PR TITLE
Change usage of [NSArray lastObject] in NSOperationQueue to be atomic

### LINKED ISSUE #2751: NSOperation_OrderOfOperations_SamePriority test can spuriously fail

### PR DESCRIPTION
**Pull request by ms-jihua**
Fixes #2751

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/2752)

### CODE REVIEW

**Comment by ms-jihua** [comment]
lastObject was the troublemaker. After this change, repro rate of test failure ~0.6% -> 0%

File: Frameworks/Foundation/NSArray.mm [hunk 1]
```diff
@@ -151,26 +151,30 @@ - (instancetype)initWithObjects:(id _Nonnull const*)objs count:(NSUInteger)count
...
+    }
 }
 
 /**
  @Status Interoperable
 */
 - (NSObject*)firstObject {
-    int count = [self count];
+    @synchronized(self) {
+        int count = [self count];
```
Comment (rajsesh-msft) [inline Frameworks/Foundation/NSArray.mm]: nit: could simplify without the local variable here.

Comment (ms-jihua) [inline Frameworks/Foundation/NSArray.mm]: might as well keep them in line

**Comment by ms-jihua** [comment]
oh right, you have a point there

**Pull request merged**